pkg/registry/service/allocator/bitmap.go

```diff
@@ -66,6 +67,30 @@ func NewAllocationMap(max int, rangeSpec string) *AllocationBitmap {
 	return &a
 }
 
+// NewContiguousAllocationMap creates an allocation bitmap using the contiguous scan strategy.
+func NewContiguousAllocationMap(max int, rangeSpec string) *AllocationBitmap {
+	a := AllocationBitmap{
+		strategy:  contiguousScanStrategy,
+		allocated: big.NewInt(0),
+		count:     0,
+		max:       max,
+		rangeSpec: rangeSpec,
+	}
+	return &a
+}
+
+// NewRandomAllocationInterface creates an allocation bitmap satisfying Interface using the
+// random scan strategy.
+func NewRandomAllocationInterface(max int, rangeSpec string) Interface {
```

Comment:
These methods implement allocator.Factory directly so you don't have to wrap the method.  Convenience.

> On May 28, 2015, at 1:11 AM, Justin Santa Barbara <notifications@github.com> wrote:
> 
> In pkg/registry/service/allocator/bitmap.go:
> 
> > @@ -66,6 +67,30 @@ func NewAllocationMap(max int, rangeSpec string) *AllocationBitmap {
> >  	return &a
> >  }
> >  
> > +// NewContiguousAllocationMap creates an allocation bitmap using the contiguous scan strategy.
> > +func NewContiguousAllocationMap(max int, rangeSpec string) *AllocationBitmap {
> > +	a := AllocationBitmap{
> > +		strategy:  contiguousScanStrategy,
> > +		allocated: big.NewInt(0),
> > +		count:     0,
> > +		max:       max,
> > +		rangeSpec: rangeSpec,
> > +	}
> > +	return &a
> > +}
> > +
> > +// NewRandomAllocationInterface creates an allocation bitmap satisfying Interface using the
> > +// random scan strategy.
> > +func NewRandomAllocationInterface(max int, rangeSpec string) Interface {
> Do we need these two ...Interface functions?
> 
> —
> Reply to this email directly or view it on GitHub.
>